nasm/private/rules/library.bzl

```diff
@@ -25,35 +27,49 @@ def nasm_assemble(
     Returns:
         File: The compiled object file.
     """
-    basename, _, _ = src.basename.rpartition(".")
-    out = ctx.actions.declare_file("{}/_obj/{}/{}.o".format(
-        ctx.label.package,
-        ctx.label.name,
-        basename,
-    ))
+    suffix = ".o"
+    if nasm_toolchain.compiler.basename.endswith(".exe"):
+        suffix = ".obj"
+
+    out_name = paths.join("_obj", ctx.label.package, ctx.label.name, src.basename + suffix)
```

Comment:
Why is `paths` needed? I also noticed `normalize` below, I don't think this should allow for `../` in the paths so perhaps that can be removed to simplify the implementation?

Reply:
It's just an additional function being imported that is designed to handle more than what is actually needed here but it's fine.

Something that was changed though is `ctx.label.package` is no longer the root. This should be fixed as the previous method was intended to match `rules_cc`.
https://github.com/bazelbuild/bazel/blob/f267da2e65931de01654398dcd64abbd6611c67b/src/main/starlark/builtins_bzl/common/cc/compile/compile_action_templates.bzl#L293-L302